CRM/Profile/Form.php

```diff
@@ -500,7 +500,7 @@ public function setDefaultsValues() {
             ) {
               $entityId = $this->_activityId;
             }
-            $url = CRM_Core_BAO_CustomField::getFileURL($entityId, $key);
+            $url = CRM_Core_BAO_CustomField::getFileURL($entityId, $key, $value);
```

Comment:
I was able to replicate the issue even after this patch. Maybe, even if `$value` is passed as NULL, `getFileURL()` returns the first file attached to the entity if multiRecordWhereClause is not being passed to the function. 

As we are setting defaults here, I think we should check for $value being set before calling to return the url. Something like:

    $url = '';
    if (isset($value)) {
      $url = CRM_Core_BAO_CustomField::getFileURL($entityId, $key, $value);
    }

It works for me. Can you please try this and check if it works for you as well ?